airflow/metrics/validators.py

```diff
@@ -155,26 +155,11 @@ def stat_name_otel_handler(
     if not (isinstance(stat_name, str) and isinstance(stat_prefix, str)):
         raise InvalidStatsNameException("Stat name and prefix must both be strings.")
 
-    if len(proposed_stat_name) > OTEL_NAME_MAX_LENGTH:
-        # If the name is in the exceptions list, do not fail it for being too long.
-        # It may still be deemed invalid for other reasons below.
-        for exemption in BACK_COMPAT_METRIC_NAMES:
```

Comment:
My IDE is currently borked so ik can't double check this easily; is BACK_COMPAT_METRIC_NAMES used anywhere else?  It's possible that can also be pruned out now if it was only ever used as a work-around for the name length bug.

Reply:
Looking at this now, I am pretty sure this was correct.  It looks like it was only ever used there and in the related tests.   Please double check then kill it if that's right.